--- a/src/resources/lib/modules/updates.py
+++ b/src/resources/lib/modules/updates.py
@@ -570,7 +570,7 @@
                         self.update_in_progress = True
                         self.do_autoupdate(None, True)
                     else:
-                        if self.oe.BUILD == 'official':
+                        if self.oe.BUILD == 'official' and self.struct['update']['settings']['UpdateNotify']['value'] == '1':
                             ceUpdate = xbmcgui.Dialog().yesno('CoreELEC', 'An update is available, would you like to download it now?')
                             if(ceUpdate):
                                 self.update_in_progress = True
