--- a/xbmc/video/dialogs/GUIDialogVideoInfo.cpp	Mon Jan 19 23:26:32 1970
+++ b/xbmc/video/dialogs/GUIDialogVideoInfo.cpp	Mon Jan 19 23:26:32 1970
@@ -1620,6 +1620,13 @@
     if (URIUtils::HasSlashAtEnd(strDeletePath))
       item->m_bIsFolder = true;
 
+    // Dont leave almost empty movie folders, remove whole movie directory
+    else
+    {
+      strDeletePath = item->GetVideoInfoTag()->m_basePath;
+      item->m_bIsFolder = true;
+    }
+	
     // check if the file/directory can be deleted
     if (CUtil::SupportsWriteFileOperations(strDeletePath))
     {
